MyExample/FormSkiCompetition.cs

```diff
@@ -35,26 +36,46 @@ public List<Skier> GetSkiers()
 
         private void buttonResults_Click(object sender, EventArgs e)
         {
-
+            var id = int.Parse(listBox1.GetItemText(listBox1.SelectedValue));
             Random random = new Random();
             var start = TimeSpan.FromSeconds(20);
             var end = TimeSpan.FromMinutes(2);
             var difference = (int)(end.TotalMilliseconds - start.TotalMilliseconds);
 
             List<Skier> all = dataProvider.GetCompetitors();
 
-
+            
             foreach (var item in all)
             {
                 var randomTime = start + TimeSpan.FromMilliseconds(random.Next(difference));
-                dataProvider.InsertResults(item.ID, randomTime, DateTime.Now.ToString("MM/dd/yyyy HH:mm:ss"));
+                
+                dataProvider.InsertResults(item.ID, randomTime, DateTime.Now.ToString("MM/dd/yyyy HH:mm:ss"), id);
             }
 
         }
 
+        public void ListBox()
+        {
+            Random random = new Random();
+            var start = TimeSpan.FromSeconds(20);
+            var end = TimeSpan.FromMinutes(2);
+            var difference = (int)(end.TotalMilliseconds - start.TotalMilliseconds);
+
+            List<Skier> all = dataProvider.GetCompetitors();
+
+
+            foreach (var item in all)
+            {
+                var randomTime = start + TimeSpan.FromMilliseconds(random.Next(difference));
+                
+                var id = int.Parse(listBox1.GetItemText(listBox1.SelectedValue));
+                dataProvider.InsertResults(item.ID, randomTime, DateTime.Now.ToString("MM/dd/yyyy HH:mm:ss"),id);
+            }
+        }
+        
         private void buttonRegister_Click(object sender, EventArgs e)

```

Comment:
Start with upper case, every method.

Reply:
move *_Click methods in their own region.

Reply:
> move *_Click methods in their own region.

 More like Command methods.